Makefile

```diff
@@ -3,62 +3,132 @@
 #                                                         :::      ::::::::    #
 #    Makefile                                           :+:      :+:    :+:    #
 #                                                     +:+ +:+         +:+      #
-#    By: mcourtoi <mcourtoi@student.42.fr>          +#+  +:+       +#+         #
+#    By: jodufour <jodufour@student.42.fr>          +#+  +:+       +#+         #
 #                                                 +#+#+#+#+#+   +#+            #
 #    Created: 2023/11/24 19:29:43 by mcourtoi          #+#    #+#              #
-#    Updated: 2024/03/01 22:03:28 by mcourtoi         ###   ########.fr        #
+#    Updated: 2024/03/06 02:23:58 by jodufour         ###   ########.fr        #
 #                                                                              #
 # **************************************************************************** #
 
 ######################################
 #              COMMANDS              #
 ######################################
-   AR = ${shell which ar} rcs
-  CXX = ${shell which c++}
- LINK = ${shell which clang++}
-MKDIR =	${shell which mkdir} -p
-   RM = ${shell which rm} -rf
+   AR := ${shell which ar} rcs
```

Comment:
I switched from operator `=` to operator `:=` because after talking about it with a friend from 42, I realized that the operator `:=` is more efficient than the operator `=`. Indeed, the operator `=` use more memory and instructions under the hood, because when you assign a variable using the operator `=`, for example:
```make
ORIGIN = foo
  COPY = ${ORIGIN}
```
then `make` will keep track that the variable `COPY` depends on the variable `ORIGIN`.
It means that if you do the following:
```make
ORIGIN = foo
  COPY = ${ORIGIN}
...
ORIGIN = bar
```
then the variable `COPY` will have the value `bar` in.
On the other hand, using the operator `:=` allows you to assign a variable without keeping track of the possible dependencies.
It means that if you do the following:
```make
ORIGIN = foo
  COPY := ${ORIGIN}
...
ORIGIN = bar
```
then the variable `COPY` will have the value that it had at the moment of its last assignation, which is `foo`.

Therefore, using the operator `:=` as much as possible is a better practice because it costs less memory and instructions under the hood.

Reply:
Thank you for the explanation, it does seem more efficient to use := for the Make